File: 1779/Solution.java
class Solution {
    public int nearestValidPoint(int x, int y, int[][] points) {
        int ind =-1, min = Integer.MAX_VALUE;
        for(int i = 0; i < points.length; i++){
            if(points[i][0] == x || points[i][1] == y){
                int cur = Math.abs(points[i][0] - x) + Math.abs(points[i][1] - y);
                if(cur < min){
                    min = cur;
                    ind = i;
                }
            }
        }
        return ind;
    }
}

File: 1779/Solution.go
func nearestValidPoint(x int, y int, points [][]int) int {
    res := math.MaxInt
    ind := -1
    for i, a := range points{
        if a[0] == x || a[1] == y{
            cur := int(math.Abs(float64(x - a[0])) + math.Abs(float64(y - a[1])))
            if cur < res{
                res = cur
                ind = i
            }
        }
    }
    return ind
}
